### DISCUSSION TITLE
Authentication Exception Previously handled with a json response

### DISCUSSION THREAD

**Discussion by aeniodunmo-vertex** [filamentphp/filament#2941] [Help]
### Package

filament/filament

### Package Version

v2.0.0

### Laravel Version

v9.0.0

### Livewire Version

v2.0.0

### PHP Version

PHP 8.1.7

### Bug description

When using filament with an API project, if Authentication Exception is handled with a json response any call by filament with that exception returns a json response so when trying to access /admin i got a json response of unauthorized which was written for my code.

### Steps to reproduce

```
<?php

namespace App\Exceptions;

use Throwable;
use ErrorException;
use Illuminate\Database\QueryException;
use Illuminate\Database\Eloquent\ModelNotFoundException;
use Illuminate\Foundation\Exceptions\Handler as ExceptionHandler;
use Illuminate\Auth\AuthenticationException as AuthenticationException;

class Handler extends ExceptionHandler
{
    /**
     * A list of the exception types that are not reported.
     *
     * @var array
     */
    protected $dontReport = [
        //
    ];

    /**
     * A list of the inputs that are never flashed for validation exceptions.
     *
     * @var array
     */
    protected $dontFlash = [
        'current_password',
        'password',
        'password_confirmation',
    ];

        /**
     * Report or log an exception.
     *
     * @param  \Throwable  $exception
     * @return void
     *
     * @throws \Throwable
     */
    public function report(Throwable $exception)
    {
        parent::report($exception);
    }

    /**
     * Register the exception handling callbacks for the application.
     *
     * @return void
     */
    // public function register()
    // {
    //     $this->reportable(function (Throwable $e) {
    //         //
    //     });
    // }

    public function render($request, Throwable $exception)
    {
        if ($exception instanceof AuthenticationException) {
            return response()->json([
                'status' => 'failed',
                'status_code' => '401',
                'message' => "You are not authorized"
            ], 401);
        }elseif($exception instanceof ModelNotFoundException)
        {
            return response()->json([
                'status' => 'failed',
                'status_code' => '404',
                'message' => "Resource not found"
            ], 404);
        }
        elseif($exception instanceof QueryException)
        {
            return response()->json([
                'status' => 'failed',
                'status_code' => '500',
                'message' => $exception->getMessage(),
                'file' => $exception->getFile(),
                'line' => $exception->getLine(),
            ], 500);
        }
            return parent::render($request, $exception);
    }
}

```

### Relevant log output

```shell
{"status":"failed","status_code":"401","message":"You are not authorized"}
```

**Comment by danharrin** [comment]
I don't know how you expect us to fix this?

**Comment by danharrin** [comment]
Use `$request` to check if the request is to the API, before handling the exceptions.

**Reply by Mustafa-KH-Hamad** [reply]
thank you this Worked for me

`  ->withExceptions(function (Exceptions $exceptions) {
        $exceptions->render(function (Throwable $e, Request $request) {
            if(!str_contains($request->url(), 'api')){
                return ;
            }
            $className = get_class($e);
            $handlers = ApiExceptions::$handlers;

            if (array_key_exists($className, $handlers)) {
                $method = $handlers[$className];
                return ApiExceptions::$method($e, $request);
            }

            return response()->json([
                'error' => [
                    'type' => basename(get_class($e)),
                    'status' => intval($e->getCode()), // returns 0 if no code
                    'message' =>  $e->getMessage()
                ]
            ]);
        });
    })->create();`
